### PR TITLE
Vmprof firefox converter blog

### PR DESCRIPTION
**Pull request by Cskorpion**

### CODE REVIEW

File: posts/2024/05/vmprof-firefox.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+<!--
+.. title: vmprof-firefox-converter
```
Comment (cfbolz) [inline posts/2024/05/vmprof-firefox.md:2]: We need a better title for the post. `vmprof-firefox-converter` should be the `slug:` (one line down).

File: posts/2024/05/vmprof-firefox.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+VMProf's console output can already give some insights into where your code spends time, 
+but it is far from showing all the information captured while profiling.
+
+There have been some tools around to visualize VMProf's output.
+Unfortunately the vmprof.com user interface is no longer available and vmprof-server is not as easy to use, you may want to take a look at a local viewer or converter.
+Those so far could give you some general visualizations of your profile. Those tools did not show any PyPy related context like PyPy's log output (PyPyLog).
+
+To bring all of those features together in one tool, you may take a look at the vmprof-firefox-converter.
+
+Created in the context of my bachelor's thesis, the vmprof-firefox-converter is a tool for analyzing VMProf profiles with the Firefox profiler user interface.
```
Comment (cfbolz) [inline posts/2024/05/vmprof-firefox.md:26]: Maybe add a sentence like "This allows us to re-use the careful user interface work that Mozilla has put into the Firefox profiler."?

Comment (cfbolz) [inline posts/2024/05/vmprof-firefox.md:26]: also, please add a link to the firefox profiler here.

File: posts/2024/05/vmprof-firefox.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+There have been some tools around to visualize VMProf's output.
+Unfortunately the vmprof.com user interface is no longer available and vmprof-server is not as easy to use, you may want to take a look at a local viewer or converter.
+Those so far could give you some general visualizations of your profile. Those tools did not show any PyPy related context like PyPy's log output (PyPyLog).
+
+To bring all of those features together in one tool, you may take a look at the vmprof-firefox-converter.
+
+Created in the context of my bachelor's thesis, the vmprof-firefox-converter is a tool for analyzing VMProf profiles with the Firefox profiler user interface. 
+The Firefox profiler offers a timeline where you can zoom into profiles and work with different visualizations like a flame graph or a stack chart.
+To understand why there is time spent inside a function, you can revisit the source code and even dive into the intermediate representation of functions executed by PyPy's just-in-time compiler.
+Additionally, there is a visualization for PyPy's log output, to keep track whether PyPy spent time inside the interpreter, JIT or GC throughout the profiling time.
```
Comment (cfbolz) [inline posts/2024/05/vmprof-firefox.md:29]: We need a new paragraph along the lines of: "In this blog post I want to show an example of how to use `vmprof-firefox-converter` for a simple Python program."

File: posts/2024/05/vmprof-firefox.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+You may already know where this comes from.
+We read a 64 Kb chunk of data from std in and pass that to ```counter.update()```, so both these tasks do more work and take longer.
+Bigger chunks mean there is less alternating between reading and counting, so there is more time spent doing work than "doing" loop iterations.
+
+
+## Getting started
+
+Both VMProf and the vmprof-firefox-converter were created for profiling PyPy, but you can also use them with CPython. 
+
+You can get the converter from [GitHub](https://github.com/Cskorpion/vmprof-firefox-converter)
```
Comment (cfbolz) [inline posts/2024/05/vmprof-firefox.md:119]: Maybe something like "The project is still somewhat experimental, so please try it out and let us know whether it worked for you!"

**Comment by Cskorpion** [comment]
Thanks for the Review. Is there anything else I can improve?

**Comment by cfbolz** [comment]
@mattip would you maybe have time to give this a read too? if you don't have time I would just click merge :blush:

File: posts/2024/05/vmprof-firefox.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+Bigger chunks mean there is less alternating between reading and counting, so there is more time spent doing work than "doing" loop iterations.
+
+
+## Getting started
+
+Both VMProf and the vmprof-firefox-converter were created for profiling PyPy, but you can also use them with CPython. 
+
+This project is still somewhat experimental, so if you want to try it out, please let us know whether it worked for you.
+
+You can get the converter from [GitHub](https://github.com/Cskorpion/vmprof-firefox-converter)
```
Comment (mattip) [inline posts/2024/05/vmprof-firefox.md:123]: Reordering.

```suggestion
You can get the converter from [GitHub](https://github.com/Cskorpion/vmprof-firefox-converter).

Both VMProf and the vmprof-firefox-converter were created for profiling PyPy, but you can also use them with CPython. 

This project is still somewhat experimental, so if you want to try it out, please let us know whether it worked for you.


```

**Review by mattip — APPROVED** [review]
LGTM. Some trivial suggestions, feel free to ignore and merge as-is.

File: posts/2024/05/vmprof-firefox.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+.. category:
+.. link:
+.. description:
+.. type: text
+.. author: Christoph Jung
+-->
+
+## Introduction
+
+If you ever wanted to profile your Python code on PyPy, you probably came across VMProf — a statistical profiler for PyPy.
```
[the remark this answers is not in the document]
Comment (Cskorpion) [reply posts/2024/05/vmprof-firefox.md:15]: I guess there should be a link to vmprof. Even though vmprof.com is no more, there is still lots of information.

File: posts/2024/05/vmprof-firefox.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+VMProf's console output can already give some insights into where your code spends time, 
+but it is far from showing all the information captured while profiling.
+
+There have been some tools around to visualize VMProf's output.
+Unfortunately the vmprof.com user interface is no longer available and vmprof-server is not as easy to use, you may want to take a look at a local viewer or converter.
+Those so far could give you some general visualizations of your profile. Those tools did not show any PyPy related context like PyPy's log output (PyPyLog).
```
[the remark this answers is not in the document]
Comment (Cskorpion) [reply posts/2024/05/vmprof-firefox.md:23]: There is a documentation: https://rpython.readthedocs.io/en/latest/logging.html

**Comment by mattip** [comment]
Thanks @Cskorpion, and thanks for the convertor!

**Pull request merged**